src/uri/UriValidator.cpp

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2023 General Motors GTO LLC
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include "UriValidator.h"
+#include "UUri.h"
+#include "UStatus.h"
+#include "google/protobuf/stubs/status.h"
+#include "UResource.h"
+#include "LongUriSerializer.h"
+#include <string.h>
+
+using namespace uprotocol::uri;
+using namespace uprotocol::utransport;
+
+/**
+ * isValidUri a {@link uri} to ensure that it has at least a name for the
+ * uEntity.
+ *
+ * @param uri {@link uri} to validate.
+ * @return Returns bool containing a true or a false
+ */
+bool UriValidator::valid_uri(const std::string& uri) {
+    auto uri_view = LongUriSerializer::deserialize(uri);
```

Comment:
Same comment as header file. We should not be serializing and deserializing in the validator, only validating the UUri object